plugins/auth-backend/src/service/OidcRouter.ts

```diff
@@ -68,6 +98,318 @@ export class OidcRouter {
       res.json(userInfo);
     });
 
+    if (
+      this.config.getOptionalBoolean(
+        'auth.experimentalDynamicClientRegistration.enabled',
+      )
+    ) {
+      // Authorization endpoint
+      // https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest
+      // Handles the initial authorization request from the client, validates parameters,
+      // and redirects to the Authorization Session page for user approval
+      router.get('/v1/authorize', async (req, res) => {
+        // todo(blam): maybe add zod types for validating input
+        const {
+          client_id: clientId,
+          redirect_uri: redirectUri,
+          response_type: responseType,
+          scope,
+          state,
+          nonce,
+          code_challenge: codeChallenge,
+          code_challenge_method: codeChallengeMethod,
+        } = req.query;
+
+        if (!clientId || !redirectUri || !responseType) {
+          this.logger.error(`Failed to authorize: Missing required parameters`);
+          return res.status(400).json({
+            error: 'invalid_request',
+            error_description:
+              'Missing required parameters: client_id, redirect_uri, response_type',
+          });
+        }
+
+        try {
+          const result = await this.oidc.createAuthorizationSession({
+            clientId: clientId as string,
+            redirectUri: redirectUri as string,
+            responseType: responseType as string,
+            scope: scope as string,
+            state: state as string,
+            nonce: nonce as string,
+            codeChallenge: codeChallenge as string,
+            codeChallengeMethod: codeChallengeMethod as string,
+          });
+
+          // todo(blam): maybe this URL could be overridable by config if
+          // the plugin is mounted somewhere else?
+          // support slashes in baseUrl?
+          const authSessionRedirectUrl = new URL(
+            `/oauth2/authorize/${result.id}`,
+            this.appUrl,
```

Comment:
```suggestion
            `./oauth2/authorize/${result.id}`,
            ensureTrailingSlash(this.appUrl),
```

or something like that to support instances that are exposed on subroutes. Or was there some spec-based reason that this absolutely HAD to be immediately under a domain?